### PR TITLE
Rstats library

### PR DESCRIPTION
**Pull request by edavidaja**
I'm still puzzling through the regex handling for R manifests, which may make sense to handle via a separate plugin, as with Ruby or Rust, but this covers scripts and RMarkdown files.

### Checklist:

- [x] If this PR is a new feature, please provide at least one example link
- [x] Make sure all of the significant new logic is covered by tests

### CODE REVIEW

**Comment by stefanbuck** [comment]
Tests are failing, because we download the markup from https://github.com/OctoLinker/testrepo/issues/2 which was missing a R-snippet (I just added that). If you run `sh ./packages/blob-reader/scripts/update-fixtures.sh` it will download all fixtures file. Please reject all changes expect `packages/blob-reader/fixtures/github.com/issue/code.html`. This should make the test pass

**Review by stefanbuck — COMMENTED** [review]
Thanks for taking the time to add R support to OctoLinker. This looks really good already. Let me know once you're done with all the regex stuff. Regarding the manifest file, I agree this should be a different plugin.

**Comment by edavidaja** [comment]
It looks like the e2e tests are timing out? Apologies for blowing up your inbox--I should have run some of those tests locally first.

File: e2e/fixtures/r/foo.R [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# @octoLinkerResolve(https://github.com/r-lib/rlang)
```
Comment (xt0rted) [inline e2e/fixtures/r/foo.R:1]: @stefanbuck are these case sensitive?

Comment (stefanbuck) [reply e2e/fixtures/r/foo.R:1]: Yes, good spot

**Comment by stefanbuck** [comment]
You might need to rebase your branch to fix the E2E test timeout. 

@xt0rted the branch is 10 commits behind OctoLinker's `main` branch which seems to be the issue as discussed a few days ago https://github.com/OctoLinker/OctoLinker/pull/1084/checks?check_run_id=1244296101#step:10:203

File: e2e/fixtures/r/foo.R (hunk 1 above)
Comment (stefanbuck) [reply e2e/fixtures/r/foo.R:1]: @edavidaja please update all OctoLinker annotations to ensure E2E tests are running

```diff
- @OctoLinkerResolve
+ @OctoLinkerResolve
```

**Comment by xt0rted** [comment]
@stefanbuck take a look at https://github.com/OctoLinker/OctoLinker/pull/1072#discussion_r502846041. That change should fix this issue, it worked on that PR.

File: e2e/fixtures/r/foo.R (hunk 1 above)
Comment (edavidaja) [reply e2e/fixtures/r/foo.R]: fixed

**Pull request merged**